### PR TITLE
AKA clean shutdown

### PR DESCRIPTION
**Pull request by areed**
aka.service with an ExecStop script to remove all pods with PVCs attached. Then poll until `lsblk` reports there are no more `rbd` devices on the node.

Still need to add support for non-masters, possibly by calling the flexvolume plugin directly on the host before the rook agent container goes down.

### CODE REVIEW

**Comment by areed** [comment]
Actually worker nodes have `/etc/kubernetes/kubelet.conf` with limited authorization, but allowed to cordon self and delete pods assigned to it. That will be sufficient.

**Comment by areed** [comment]
Always stop app on master shutdown to avoid race condition in shared fs mounts on bootup.

File: install_scripts/templates/common/kubernetes.sh [hunk 1]
```diff
@@ -1762,3 +1762,86 @@ checkDockerK8sVersion()
...
+done
+
+while \$(cat /proc/mounts | grep -q ':6789:/'); do
+        echo "Waiting for Ceph shared filesystems to unmount"
+        sleep 1
+done
+
+EOF
+
+chmod a+x /opt/replicated/shutdown.sh
```
Comment (emosbaugh) [inline install_scripts/templates/common/kubernetes.sh]: could this be less permissive?

File: install_scripts/templates/common/kubernetes.sh [hunk 2]
```diff
@@ -1762,3 +1762,86 @@ checkDockerK8sVersion()
...
+# wait for Kubernets API
+master=\$(cat /etc/kubernetes/kubelet.conf | grep server | awk '{ print \$2 }')
+while [ "\$(curl --noproxy "*" -sk \$master/healthz)" != "ok" ]; do
+        sleep 1
+done
+
+KUBECONFIG=/etc/kubernetes/kubelet.conf kubectl uncordon \$(hostname)
+EOF
+
+chmod a+x /opt/replicated/start.sh
```
Comment (emosbaugh) [inline install_scripts/templates/common/kubernetes.sh]: could this be less permissive?

**Pull request merged**